Turkey/Turkey.csproj

```diff
@@ -2,7 +2,7 @@
 
   <PropertyGroup>
     <OutputType>Exe</OutputType>
-    <TargetFramework>net6.0</TargetFramework>
+    <TargetFramework>net8.0</TargetFramework>
```

Comment:
thought(non-blocking): I am not sure if this is necessary. 

note: You can already overwrite this with

```bash
export TargetBundledFramework='true'
export BundledNETCoreAppTargetFrameworkVersion='8.0'
```

See: https://github.com/canonical/dotnet-test-runner/blob/c988ad2b1c64a4ff5459992a5f682b3d6f9a79d3/Turkey/Turkey.csproj#L9

Reply:
note: Additionally you can technically use the .NET 6 produced `.dll` and run it with .NET 8. Therefore there is no technical need to change this now.

Reply:
In retrospect, I think I'll revert this. The idea was to keep the runner up-to-date with latest language features, if we ever use them. But considering this is also shipping in .NET 6 packages and will do so for some time, I'd prefer to keep this `net6.0`.

Reply:
Setting this to resolved with e3f9d37.